### PR TITLE
Aic bic

### PR DESCRIPTION
**Pull request by mgross**
Introduce new UI tab: AIC + BIC

### CODE REVIEW

**Comment by mgross** [comment]
@arunge : can you have a look it?

**Review by arunge — CHANGES_REQUESTED** [review]
@mgross When trying to load an online test model (1) the app crashes (2). Can you please check what is going on?

image

image

**Comment by mgross** [comment]
fixed it

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: PlotR
 Title: Plot things
-Version: 23.05.2.1
+Version: 23.06.1.0
```
Comment (arunge) [inline DESCRIPTION]:
```suggestion
Version: 23.06.0
```

File: R/01-fitModel.R [hunk 2]
```diff
@@ -144,6 +144,8 @@ fitPlotRModelMC <- function(data,
   res$beta <- do.call("rbind", lapply(1:length(ret), function(x) ret[[x]]$beta))
   res$betaSigma <- do.call("rbind", lapply(1:length(ret), function(x) ret[[x]]$betaSigma))
   res$sigma <- do.call("c", lapply(1:length(ret), function(x) ret[[x]]$sigma))
+  res$llog <- do.call("cbind", lapply(1:length(ret), function(x) ret[[x]]$llog))
```
Comment (arunge) [inline R/01-fitModel.R:147]: Is it correct to use here `cbind` and not `rbind` like for `$beta` and for `$betaSigma`?

File: R/01-fitModel.R [hunk 3]
```diff
@@ -358,24 +360,33 @@ fitPlotRModel <- function(data,
   usedsamples <- seq(from = burnin, to = iter, by = every)
   if (sdVar){
     seTotal <- range(sqrt(apply(sapply(1:length(usedsamples), function(x)
-      (XX2 %*% betamc[usedsamples[x], ]) * sRe + mRe), 1, var) +
+      (XX %*% betamc[usedsamples[x], ]) * sRe + mRe), 1, var) +
```
Comment (arunge) [inline R/01-fitModel.R]: I do not understand this part. What is the difference between `XX` and `XX2`?
Why do you not use `XX2` anymore as you did before? Why the switch to `XX`?

**Review by arunge — CHANGES_REQUESTED** [review]
- [x] I opened a new PR where I extracted the function to create the table and added a test. 
  Please merge [that](https://github.com/Pandora-IsoMemo/plotr/pull/38) first.
- [x] I have some questions regarding `XX`
  - Is it normal that the values in the table differ that much if variable standard deviation (m1) or no variable standard deviation (ms) is used?
  image

**Comment by arunge** [comment]
@mgross Please also check and merge this PR first: https://github.com/Pandora-IsoMemo/plotr/pull/39 where I extracted the log likelihood computation.

In general, the code already has a lot of long if/else conditionals and is very hard to read and to understand. Could you please avoid to introduce more conditionals directly into an existing function? Please try to split things and create new functions with meaningful names to add new functionality.

**Comment by arunge** [comment]
@mgross Maybe this got lost: Could check the questions in:
- https://github.com/Pandora-IsoMemo/plotr/pull/37#pullrequestreview-1504642828
- https://github.com/Pandora-IsoMemo/plotr/pull/37#discussion_r1246184791
- https://github.com/Pandora-IsoMemo/plotr/pull/37#discussion_r1246186319

Afterwards we can continue here. :slightly_smiling_face:

File: R/01-fitModel.R (hunk 2 above)
Comment (mgross) [reply R/01-fitModel.R:147]: Yes, this is correct :)

File: R/01-fitModel.R (hunk 3 above)
Comment (mgross) [reply R/01-fitModel.R]: You're right. XX2 is the right matrix to use here, I got confused myself. XX2 is the matrix on the original X-Values, while XX is the matrix on the changed X-Values (if there is uncertainty in X)

Comment (mgross) [reply R/01-fitModel.R]: This should also fix the large AIC/BIC differences between models with and without uncertainty/standard deviation in X

**Comment by mgross** [comment]
@arunge:
I answered your questions and fixed one issue

File: R/01-fitModel.R (hunk 3 above)
Comment (mgross) [reply R/01-fitModel.R]: In my tests, all is fine now, AIC and BIC are in that case as you would expect (some differences can and will still occur, especially for smaller data)

**Review by arunge — APPROVED** [review]
Thanks for your comments! No all is more clear.

**Pull request merged**